frontend/src/components/admin/AssignedStoryTranslationsTable.tsx

```diff
@@ -0,0 +1,176 @@
+import React, { useState } from "react";
+import { Icon } from "@chakra-ui/icon";
+import { MdDelete } from "react-icons/md";
+import {
+  Badge,
+  Button,
+  Table,
+  Thead,
+  Tbody,
+  Tr,
+  Th,
+  Td,
+  IconButton,
+} from "@chakra-ui/react";
+import { StoryTranslation } from "../../APIClients/queries/StoryQueries";
+import { getLevelVariant } from "../../utils/StatusUtils";
+import convertStageTitleCase from "../../utils/StageUtils";
+import { convertLanguageTitleCase } from "../../utils/LanguageUtils";
+
+export type AssignedStoryTranslationsTableProps = {
+  storyTranslations: StoryTranslation[];
+  setStoryTranslations: (newStoryTranslations: StoryTranslation[]) => void;
+  userId: number;
+};
+
+const AssignedStoryTranslationsTable = ({
+  storyTranslations,
+  setStoryTranslations,
+  userId,
+}: AssignedStoryTranslationsTableProps) => {
+  const [isAscendingTitle, setIsAscendingTitle] = useState(true);
+  const [isAscendingRole, setIsAscendingRole] = useState(true);
+  const [isAscendingLanguage, setIsAscendingLanguage] = useState(true);
+  const [isAscendingStage, setIsAscendingStage] = useState(true);
+
+  const getRole = (translation: StoryTranslation) => {
```

Comment:
all this sorting stuff reaaaally lengthens the file length. Thoughts on making a new `utils/` file and moving all this sort stuff to it?

Reply:
Hmm the sorting functions currently use a state variable `isAscending...` and also use the function `getRole` on the `roleSort` function. Moving all of these to a utils folder would be a bit messy. I think it'll look better in here